src/asf_enumeration/aria_s1_gunw.py

```diff
@@ -229,36 +229,59 @@ def get_acquisition(frame: int | AriaFrame, date: datetime.date) -> Sentinel1Acq
     return acquisition
 
 
-def product_exists(frame: int | AriaFrame, reference_date: datetime.date, secondary_date: datetime.date) -> bool:
+def product_exists(reference_date: datetime.date | str, secondary_date: datetime.date | str, frame_id: int) -> bool:
     """Check if ARIA product already exists.
 
     Args:
-        frame: ARIA frame ID or frame object
-        reference_date: Reference date of the product
-        secondary_date: Secondary date of the product
+        reference_date: Reference date of the product as a `datetime.date` object or a date string in ISO format
+        secondary_date: Secondary date of the product as a `datetime.date` object or a date string in ISO format
+        frame_id: ARIA frame ID
 
     Returns:
-        exists_in_archive: whether the product already exists in ASF's archive
+        Whether the product already exists in ASF's archive
+
+    """
+    return get_product(reference_date, secondary_date, frame_id) is not None
+
+
+def get_product(
+    reference_date: datetime.date | str, secondary_date: datetime.date | str, frame_id: int
+) -> asf.ASFProduct | None:
+    """Get the ARIA product for the given parameters, if it exists.
+
+    Args:
+        reference_date: Reference date of the product as a `datetime.date` object or a date string in ISO format
+        secondary_date: Secondary date of the product as a `datetime.date` object or a date string in ISO format
+        frame_id: ARIA frame ID
 
+    Returns:
+        The product if it exists, otherwise None.
     """
-    if isinstance(frame, AriaFrame):
-        frame = frame.id
+    if isinstance(reference_date, str):
+        reference_date = datetime.date.fromisoformat(reference_date)
+
+    if isinstance(secondary_date, str):
+        secondary_date = datetime.date.fromisoformat(secondary_date)
+
+    _validate_frame_id(frame_id)
 
-    _validate_frame_id(frame)
     date_buffer = datetime.timedelta(days=1)
 
     results = asf.search(
         dataset=asf.constants.DATASET.ARIA_S1_GUNW,
-        frame=frame,
+        frame=frame_id,
         start=reference_date - date_buffer,
         end=reference_date + date_buffer,
     )
+    results = [
+        result
+        for result in results
+        if _gunw_dates_match(result.properties['sceneName'], reference_date, secondary_date)
+    ]
+    if results:
+        return max(results, key=lambda product: product.meta['revision-date'])
```

Comment:
I would put the error case as the early return, but either works.

Reply:
Done.